### PR TITLE
Fully working IME - WASM, ios and Android

### PR DESCRIPTION
**Pull request by danwalmsley**
Combined effort: @Gillibald @kekekeks @danwalmsley 

- win32 preedit support.
- android pre edit
- ios preedit

- refactors and fixes to ime api, with working android and ios keyboard.
- wasm - sync ime context with browser.

Remaining todo:

- [x] Return key setting
- [x] Focus issue on android.
- [x] pressing button when keyboard is showing should trigger button command as well as close keyboard.
- [x] Accepts return obeyed

### CODE REVIEW

File: src/Avalonia.Base/Input/TextInput/TextInputReturnKeyType.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+{
+    Default,
+    Go,
+    Google,
+    Join,
+    Next,
+    Route,
+    Search,
+    Send,
+    Yahoo,
```
Comment (maxkatz6) [inline src/Avalonia.Base/Input/TextInput/TextInputReturnKeyType.cs:13]: Do we need something as specific as Yahoo or Google here? Wouldn't it be enough to use Search instead?

Comment (kekekeks) [reply src/Avalonia.Base/Input/TextInput/TextInputReturnKeyType.cs:13]: Where did we get those values from? Looks looks something OS-specific

Comment (danwalmsley) [reply src/Avalonia.Base/Input/TextInput/TextInputReturnKeyType.cs:13]: https://developer.apple.com/documentation/uikit/uireturnkeytype

Comment (danwalmsley) [reply src/Avalonia.Base/Input/TextInput/TextInputReturnKeyType.cs]: used a small subset that is cross platform.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024016-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024020-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024027-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by danwalmsley** [comment]
we decided to merge this as only a few remaining issues with WASM implementation, and that will be better dealt with in PR for new wasm backend.

**Pull request merged**